src/bin.rs

```diff
@@ -145,3 +175,255 @@ fn play(cfg: Play) {
+    std::thread::spawn(move || {
+        gather_telemetry_from_file(file, tx);
+    });
+
+    let mut telemetry_messages: Vec<TelemetryMessage> = Vec::new();
+
+    let mut nb_boot_messages: u32 = 0;
+    let mut nb_alarm_traps: u32 = 0;
+    let mut nb_data_snapshots: u32 = 0;
+    let mut nb_machine_state_snapshots: u32 = 0;
+    let mut nb_stopped_messages: u32 = 0;
+
+    loop {
+        match rx.try_recv() {
+            Ok(message) => {
+                match message {
+                    Ok(TelemetryMessage::BootMessage(_)) => {
+                        nb_boot_messages += 1;
+                    }
+                    Ok(TelemetryMessage::AlarmTrap(_)) => {
+                        nb_alarm_traps += 1;
+                    }
+                    Ok(TelemetryMessage::DataSnapshot(_)) => {
+                        nb_data_snapshots += 1;
+                    }
+                    Ok(TelemetryMessage::MachineStateSnapshot(_)) => {
+                        nb_machine_state_snapshots += 1;
+                    }
+                    Ok(TelemetryMessage::StoppedMessage(_)) => {
+                        nb_stopped_messages += 1;
+                    }
+                    _ => {}
+                }
+                telemetry_messages.push(message.unwrap());
+            }
+            Err(TryRecvError::Empty) => {
+                std::thread::sleep(std::time::Duration::from_millis(1));
+            }
+            Err(TryRecvError::Disconnected) => {
+                println!("Statistics");
+                println!("Nb BootMessages: {}", nb_boot_messages);
+                println!("Nb AlarmTraps: {}", nb_alarm_traps);
+                println!("Nb DataSnapshots: {}", nb_data_snapshots);
+                println!("Nb MachineStateSnapshot: {}", nb_machine_state_snapshots);
+                println!("Nb StoppedMessage: {}", nb_stopped_messages);
+                println!(
+                    "Estimated duration: {:.3} seconds",
+                    compute_duration(telemetry_messages) as f32 / 1000_f32
+                );
+                std::process::exit(0);
+            }
+        }
+    }
+}
+
+fn compute_duration(messages: Vec<TelemetryMessage>) -> u32 {
+    let mut duration: u32 = 0;
+
+    for message in &messages {
+        match message {
+            TelemetryMessage::DataSnapshot(_) => {
+                duration += 10;
+            }
+
+            TelemetryMessage::StoppedMessage(_) => {
+                duration += 100;
+            }
+            _ => {}
+        }
+    }
+
+    duration
+}
+
+#[cfg(test)]
+mod tests {
+    // Note this useful idiom: importing names from outer (for mod tests) scope.
+    use super::*;
+
+    #[test]
+    fn test_compute_duration_no_data() {
+        assert_eq!(compute_duration(vec![]), 0);
+    }
+
+    #[test]
+    fn test_compute_duration_one_boot_message() {
+        let mut vect: Vec<TelemetryMessage> = Vec::new();
+        vect.push(TelemetryMessage::BootMessage(BootMessage {
```

Comment:
This can be done in a more literal way with `let vect = vec![TelemetryMessage::BootMessage(...)];`. I didn't try it here but I feel it is better as it removes the need to mutate the vec.

_(If it works, you can also adapt the following tests.)_

Reply:
I will try your proposal.

Reply:
Done